### PR TITLE
APIv1: remove old Nessie API

### PR DESCRIPTION
**Pull request by snazy**
Update all tests that use the old Nessie API to use the Nessie v1 API.

Update nessie-deltalake to use the Nessie v1 API

Removed interfaces:
* `org.projectnessie.client.NessieClient` (the V0.9 API)
* `org.projectnessie.api.ConfigAPi`
* `org.projectnessie.api.ContentsApi`
* `org.projectnessie.api.TreeApi`

Depends on #1932

---
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/2011)

### CODE REVIEW

File: clients/deltalake/src/main/scala/org/projectnessie/deltalake/NessieLogStore.scala [hunk 1]
```diff
@@ -99,8 +100,8 @@ class NessieLogStore(sparkConf: SparkConf, hadoopConf: Configuration)
 
     try {
       val ref = Option(requestedRef)
-        .map(client.getTreeApi.getReferenceByName(_))
-        .getOrElse(client.getTreeApi.getDefaultBranch)
+        .map(api.getReference.refName(_).submit())
```
Comment (dimas-b) [inline clients/deltalake/src/main/scala/org/projectnessie/deltalake/NessieLogStore.scala:103]: nit: `.submit()` looks odd in this context... I get the impression that the result is received asynchronously. Would `.retrieve()` be more readable?

This is not really related to the removal of the old `NessieClient`, but more of a general API clean-up comment.

File: clients/deltalake/src/test/java/org/projectnessie/deltalake/ITDeltaLog.java [hunk 2]
```diff
@@ -29,19 +29,25 @@
...
+import org.projectnessie.client.http.HttpClientBuilder;
 import org.projectnessie.client.tests.AbstractSparkTest;
 import org.projectnessie.model.Branch;
-import org.projectnessie.model.ImmutableMerge;
 import org.projectnessie.model.Reference;
 import scala.Tuple2;
 
 class ITDeltaLog extends AbstractSparkTest {
 
+  NessieApiV1 api;
```
Comment (dimas-b) [inline clients/deltalake/src/test/java/org/projectnessie/deltalake/ITDeltaLog.java:49]: nit: `private`?

File: clients/deltalake/src/test/java/org/projectnessie/deltalake/ITDeltaLogBranches.java [hunk 3]
```diff
@@ -45,7 +47,7 @@
 
 class ITDeltaLogBranches extends AbstractSparkTest {
 
-  private NessieClient client;
+  static NessieApiV1 api;
```
Comment (dimas-b) [inline clients/deltalake/src/test/java/org/projectnessie/deltalake/ITDeltaLogBranches.java:50]: nit: why `static`?

File: clients/deltalake/src/test/java/org/projectnessie/deltalake/ITDeltaLog.java (hunk 2 above)
Comment (snazy) [reply clients/deltalake/src/test/java/org/projectnessie/deltalake/ITDeltaLog.java:49]: nah - it's a test

Comment (snazy) [reply clients/deltalake/src/test/java/org/projectnessie/deltalake/ITDeltaLog.java:49]: BTW: that's code from #1932 (which is why this PR is in draft)

File: clients/deltalake/src/test/java/org/projectnessie/deltalake/ITDeltaLogBranches.java (hunk 3 above)
Comment (snazy) [reply clients/deltalake/src/test/java/org/projectnessie/deltalake/ITDeltaLogBranches.java:50]: BTW: that's code from #1932 (which is why this PR is in draft)

File: clients/deltalake/src/main/scala/org/projectnessie/deltalake/NessieLogStore.scala (hunk 1 above)
Comment (snazy) [reply clients/deltalake/src/main/scala/org/projectnessie/deltalake/NessieLogStore.scala:103]: Mind opening a ticket for that to be done when everything else is merged?

Comment (nastra) [reply clients/deltalake/src/main/scala/org/projectnessie/deltalake/NessieLogStore.scala:103]: +1 on updating `sumit` to `retrieve` or `get`

File: servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java [hunk 4]
```diff
@@ -112,24 +106,36 @@
     Locale.setDefault(Locale.ENGLISH);
   }
 
-  protected void init(NessieClient client, HttpClient httpClient) {
-    this.client = client;
+  protected void init(URI uri) {
+    NessieApiV1 api =
+        HttpClientBuilder.builder().withUri(uri).build(NessieApiVersion.V_1, NessieApiV1.class);
```
Comment (nastra) [inline servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java:111]: this part should be configurable so that we can just pass NessieApiV1 from subclasses to `init` as grpc stuff will use a different builder

File: servers/quarkus-server/src/test/java/org/projectnessie/server/ITNativeRest.java [hunk 5]
```diff
@@ -33,15 +28,7 @@
   @Override
   @BeforeEach
   public void setUp() throws Exception {
-    URI uri = URI.create("http://localhost:19121/api/v1");
```
Comment (nastra) [inline servers/quarkus-server/src/test/java/org/projectnessie/server/ITNativeRest.java:36]: we need to init the NessieApiV1 here and just pass it down for grpc

File: servers/quarkus-server/src/test/java/org/projectnessie/server/TestRest.java [hunk 6]
```diff
@@ -34,15 +29,7 @@
   @Override
   @BeforeEach
   public void setUp() throws Exception {
-    URI uri = URI.create("http://localhost:19121/api/v1");
```
Comment (nastra) [inline servers/quarkus-server/src/test/java/org/projectnessie/server/TestRest.java:37]: same as above

**Review by nastra — CHANGES_REQUESTED** [review]
mostly LGTM, but we need to make sure that subclasses of AbstractRestTest init NessieApiV1 on their own and just pass it

File: servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java (hunk 4 above)
Comment (snazy) [reply servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java:111]: That's why there are 2 `init` methods:
`protected void init(URI uri)` which calls `protected void init(NessieApiV1 api, HttpClient httpClient)`

File: servers/quarkus-server/src/test/java/org/projectnessie/server/ITNativeRest.java (hunk 5 above)
Comment (snazy) [reply servers/quarkus-server/src/test/java/org/projectnessie/server/ITNativeRest.java:36]: See https://github.com/projectnessie/nessie/pull/2011#discussion_r711102482

File: servers/quarkus-server/src/test/java/org/projectnessie/server/TestRest.java (hunk 6 above)
Comment (snazy) [reply servers/quarkus-server/src/test/java/org/projectnessie/server/TestRest.java:37]: See https://github.com/projectnessie/nessie/pull/2011#discussion_r711102482

File: clients/deltalake/src/main/scala/org/projectnessie/deltalake/NessieLogStore.scala (hunk 1 above)
Comment (snazy) [reply clients/deltalake/src/main/scala/org/projectnessie/deltalake/NessieLogStore.scala:103]: In a different PR IMO. This one is updating the call-sites, not the API itself.

File: servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java (hunk 4 above)
Comment (nastra) [reply servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java:111]: ah sorry, totally missed that somehow

**Review by nastra — APPROVED** [review]

File: clients/deltalake/src/main/scala/org/projectnessie/deltalake/NessieLogStore.scala (hunk 1 above)
Comment (dimas-b) [reply clients/deltalake/src/main/scala/org/projectnessie/deltalake/NessieLogStore.scala:103]: Different PR, sure. I'm fine with `get` too.

Comment (dimas-b) [reply clients/deltalake/src/main/scala/org/projectnessie/deltalake/NessieLogStore.scala:103]: #2029

**Review by dimas-b — APPROVED** [review]
